File: app/src/main/java/com/cn29/aac/ui/viewpager/PagerActivityModule.java
package com.cn29.aac.ui.viewpager;

import com.cn29.aac.ui.viewpager.vm.PagerAgentViewModel;

import androidx.lifecycle.ViewModelProviders;
import dagger.Module;
import dagger.Provides;

/**
 * Created by Charles Ng on 3/10/2017.
 */

@Module
public class PagerActivityModule {

  @Provides
  PagerAgentViewModel providePagerAgentVm(PagerActivity pagerActivity) {

    PagerAgentViewModel pagerAgentViewModel = ViewModelProviders.of(pagerActivity)
        .get(PagerAgentViewModel.class);
    pagerAgentViewModel.init();
    return pagerAgentViewModel;
  }
}

File: app/src/main/java/com/cn29/aac/ui/viewpager/PagerActivityModule.kt
package com.cn29.aac.ui.viewpager

import androidx.lifecycle.ViewModelProvider
import com.cn29.aac.ui.viewpager.vm.PagerAgentViewModel
import dagger.Module
import dagger.Provides

@Module
class PagerActivityModule {
    @Provides
    fun providePagerAgentVm(pagerActivity: PagerActivity): PagerAgentViewModel {
        return ViewModelProvider(pagerActivity)
                .get(PagerAgentViewModel::class.java)
                .also {
                    it.init()
                }
    }
}